### PR TITLE
fix(metrics): Re-enable toggling metrics for user

### PR DESCRIPTION
**Pull request by vbudhram**
Because:

* We disabled checking for the metrics opt out property on the user in train 218 since it caused unexpected amount of db load

This commit:

* Re-enables the metrics checking in auth-server
* Caches the metricsOptOut property in the request so it only is looked up once
* I am not 100% sure if this will fix the issue since the oauth metric events now need to do the opt out lookup.

Closes https://github.com/mozilla/fxa/issues/10963

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/server.js [hunk 1]
```diff
@@ -294,6 +344,11 @@ async function create(log, error, config, routes, db, translator, statsd) {
     'setMetricsFlowCompleteSignal',
     metricsContext.setFlowCompleteSignal
   );
+  server.decorate(
+    'request',
+    'getAndSetMetricsEnabled',
+    metricsContext.getAndSetMetricsEnabled
+  );
```
Comment (dannycoates) [inline packages/fxa-auth-server/lib/server.js:351]: where is this used?

File: packages/fxa-auth-server/lib/server.js [hunk 2]
```diff
@@ -238,6 +239,55 @@ async function create(log, error, config, routes, db, translator, statsd) {
...
+      // Note that unverified accounts can not be opted out of metrics.
+
+      let uid;
+      if (
+        request.auth &&
+        request.auth.credentials &&
+        request.auth.credentials.uid
+      ) {
+        // sessionToken strategy sets this property already
+        return !request.auth.credentials.metricsOptOutAt;
```
Comment (dannycoates) [inline packages/fxa-auth-server/lib/server.js:254]: 👍

**Review by dannycoates — APPROVED** [review]

File: packages/fxa-auth-server/lib/server.js (hunk 1 above)
Comment (vbudhram) [reply packages/fxa-auth-server/lib/server.js:351]: This was left over from a different approach I was thinkign about, will remove.

File: packages/fxa-auth-server/lib/server.js [hunk 3]
```diff
@@ -238,6 +239,55 @@ async function create(log, error, config, routes, db, translator, statsd) {
...
+        } catch (err) {
+          // Unknown accounts will have the default experience
+        }
+      }
+
+      if (!uid) {
+        return true;
+      }
+
+      return Account.metricsEnabled(uid);
```
[the remark this answers is not in the document]
Comment (vbudhram) [reply packages/fxa-auth-server/lib/server.js:288]: Yea that is correct, it should only get called once on a request.

**Pull request merged**